server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -756,12 +772,16 @@ public SearchHit asUnpooled() {
             innerHits == null
                 ? null
                 : innerHits.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().asUnpooled())),
-            documentFields,
-            metaFields,
+            cloneIfHashMap(documentFields),
```

Comment:
Sorry if this is obvious, but why do we need to clone in the unpooled case?

`deallocate` won't be called at all for unpooled hits

Reply:
Ah, it's about not sharing the references with pooled instances that will now modify the state - I got there in the end :)